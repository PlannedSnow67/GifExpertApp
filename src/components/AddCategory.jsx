import { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState("");

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue.trim());
        setInputValue("");
    }
    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input
                type="text"
                placeholder="Search"
                value={inputValue}
                onChange={(e) => setInputValue(e.target.value)}
            />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
