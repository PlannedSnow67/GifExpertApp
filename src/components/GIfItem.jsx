import { PropTypes } from 'prop-types';
export const GIfItem = ({ title, url }) => {
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
}

GIfItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
