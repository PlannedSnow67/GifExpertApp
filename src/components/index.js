export * from './AddCategory';
export * from './GIfItem';
export * from './GifGrid';