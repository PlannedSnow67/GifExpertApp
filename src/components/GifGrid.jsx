import { PropTypes } from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GIfItem } from "./GIfItem";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);
    console.log(isLoading);
    return (
        <>
            <h3>{category}</h3>
            {isLoading && <h2 className="loading">Loading...</h2>}
            <div className='card-grid'>
                {
                    images.map((image) => <GIfItem key={image.id} {...image} />)
                }
            </div>

        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
