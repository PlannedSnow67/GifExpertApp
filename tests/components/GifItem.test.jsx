import { describe, test, expect } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import { GIfItem } from '../../src/components/GIfItem';

describe('GifItem testing', () => {

    const title = 'Samurai X';
    const url = 'https://localhost/algo.jpg';

    test('snapshot checking', () => {
        // make snapshot test
        const { container } = render(<GIfItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('should show image with url and alt correct', async () => {
        render(<GIfItem title={title} url={url} />)
        // screen.debug();
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('should show title in the component', () => {
        render(<GIfItem title={title} url={url} />);
        const titleElement = screen.getByText(title);
        expect(titleElement).toBeTruthy();
    })
})