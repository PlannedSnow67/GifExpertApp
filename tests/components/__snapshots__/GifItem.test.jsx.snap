// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem testing snapshot checking 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Samurai X"
      src="https://localhost/algo.jpg"
    />
    <p>
      Samurai X
    </p>
  </div>
</div>
`;
